README.md

```diff
@@ -1111,6 +1111,8 @@ Other Style Guides
   <a name="constructors--chaining"></a><a name="9.3"></a>
   - [9.3](#constructors--chaining) Methods can return `this` to help with method chaining.
 
+    > When? Use chaining when functions manipulate the internals of an object instead of just returning data.
```

Comment:
Definitely not "only use", because (for example) all the non-mutating array methods are perfect chaining candidates but do not manipulate any internals.

In general, this section may need to be reworked to suggest making classes immutable (ie, to always return a new instance, allowing chaining and also avoiding mutation). However that shouldn't go in this PR.